Add evals with Inspect guide to Inference Providers (#2036)
huggingface/hub-docs: docs/inference-providers/guides/evaluation-inspect-ai.md

To see the evaluation samples and inference in real time you can check the `Running samples` tab:

![Screenshot of running samples with gpt-oss-20b](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/evals-guide-running-samples.png)

Once it finishes, we'll see the evaluation results:

![Screenshot of theory of mind's eval results with gpt-oss-20b](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/evals-guide-first-eval-results.png).

Besides the command line report, Inspect comes with a nice viewer UI. We can launch it with the following command:

```bash
inspect viewer
```
![Screenshot of inspect viewer results with gpt-oss-20b](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/evals-guide-first-eval-viewer.png).

Nice! We have just evaluated our first model with Inspect and Inference Providers. Now let's look at more advanced examples.

## Example: Benchmarking several models for a task

> In this section, we will evaluate several models for a specific task. This is useful for selecting the most suitable model for your project and establishing a baseline if you plan to fine-tune a model for your use case. Fortunately, Inspect and Inference Providers make this task very easy. We just need to:

Reviewer: I'd also remove "Fortunately".

2. **Write and run the evaluation**. To run the evaluation across different models, there are two options: (1) run the `inspect eval` command several times with different `--model` values, and (2) write a single Python file that runs the evaluation over the list of models. In the previous section we've seen how to run Inspect with the first approach, let's see how to write a single script for the second approach.

```python
from inspect_ai import eval
from inspect_ai import Task, task
from inspect_ai.dataset import example_dataset
from inspect_ai.scorer import model_graded_fact
from inspect_ai.solver import generate

@task
def theory_of_mind():
    return Task(
        dataset=example_dataset("theory_of_mind"),
        solver=generate(),
        scorer=model_graded_fact()
    )

target_models = [
    "MiniMaxAI/MiniMax-M2",
    "openai/gpt-oss-20b",
    "openai/gpt-oss-120b", 
    "moonshotai/Kimi-K2-Instruct-0905"
]

> for model in target_models:

Author: Thanks for the pointers! I'll test and rewrite the example, mentioning both approaches.

    eval(theory_of_mind(), model=f"hf-inference-providers/{model}")
```

If we save this to a file `models_bench.py` we can run it like this:

```bash
python models_bench.py
```

If everything went well we will see the evaluation running sequentially for each model on the list. To analyze the results, we can use the viewer:

```bash
inspect viewer
```
![Screenshot of inspect viewer results with gpt-oss-20b](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/evals-guide-model-bench-viewer.png).

## Example: Comparing several inference providers for a task

> In this section, we will evaluate the same model across different providers. Inference Providers gives us access to many providers for the same model. As performance might vary across providers, this is a useful metric to select the provider, in addition to other important factors such as speed and cost.

Reviewer: ```suggestion
In this section, we will evaluate the same model across different providers. Inference Providers gives us access to many providers for the same model. Performance might vary across providers, so this is a useful factor, in addition to speed and cost, to choose the most appropriate inference provider for your task.
```

from inspect_ai.scorer import model_graded_fact
from inspect_ai.solver import generate

@task
def theory_of_mind():
    return Task(
        dataset=example_dataset("theory_of_mind"),
        solver=generate(),
        scorer=model_graded_fact()
    )

target_providers = [
    "together",
    "sambanova", 
    "groq",
    "novita",
    "nebius",
    "cerebras",
    "nscale",
    "hyperbolic",
    "fireworks-ai",
    "scaleway"
]
for provider in target_providers:
    eval(theory_of_mind(), model=f"hf-inference-providers/openai/gpt-oss-120b:{provider}")
```

Launching the viewer and sorting by score, we can compare the performance across providers, similar to the table below:

| Model | Provider | Score |
|-------|----------|-------|
| openai/gpt-oss-120b | hyperbolic | 0.84 |
| openai/gpt-oss-120b | fireworks-ai | 0.82 |
| openai/gpt-oss-120b | nscale | 0.82 |
| openai/gpt-oss-120b | nebius | 0.82 |
| openai/gpt-oss-120b | scaleway | 0.81 |
| openai/gpt-oss-120b | together | 0.81 |
| openai/gpt-oss-120b | cerebras | 0.8 |
| openai/gpt-oss-120b | novita | 0.8 |
| openai/gpt-oss-120b | groq | 0.8 |
| openai/gpt-oss-120b | sambanova | 0.8 |

> 

Reviewer 2: Maybe worth a quick note to say you can also filter using `:cheapest` and `:fastest` https://github.com/huggingface/hub-docs/pull/2025/
